cli/pkg/local/app.go

```diff
@@ -280,6 +282,12 @@ func NewApp(ctx context.Context, opts *AppOptions) (*App, error) {
 		return nil, err
 	}
 
+	// Collect and emit information about registered source types
+	err = emitSourceTelemetry(ctx, opts, rt, inst)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
This is currently a blocking operation with a network call that will hold back starting of the application. I think we should run it in a goroutine.